enterprise/server/remote_execution/runner/runner.go

```diff
@@ -168,8 +173,8 @@ func (r *CommandRunner) pullCredentials() container.PullCredentials {
 	return container.GetPullCredentials(r.env, r.PlatformProperties)
 }
 
-func (r *CommandRunner) PrepareForTask(ctx context.Context, task *repb.ExecutionTask) error {
-	r.Workspace.SetTask(task)
+func (r *CommandRunner) PrepareForTask(ctx context.Context) error {
+	r.Workspace.SetTask(r.task)
```

Comment:
since the runner can be reused, it can handle multiple tasks. Are you sure you want to be resetting the task instance variable directly all the time instead of passing the task in when you call PrepareForTask? that seems like a more natural flow...?

Reply:
So, the current flow is that when you check out a runner from the pool, you have to pass in a Task, and we pick a runner that is compatible with the Task (same container-image, same remote instance name, etc.). The runner is then "bound" to that task for as long as it is checked out from the pool -- this is already implicit in how we store the task's PlatformProperties in the runner state, and how we make use of those PlatformProperties in the various lifecycle funcs.

I think it is somewhat error prone to allow passing a Task into this function, because it'd be possible to make the mistake of checking a runner out of the pool that is able to run one task, but then prepare it for a different task, and then actually run yet another task. It seems cleaner to be able to check out a runner bound to a task, and then the caller just has the responsibility of walking that one task through its lifecycle (Prepare, then Run).

Reply:
Got it, sg. Thanks for explaining it!